Sources/GraphQLTransportWS/Server.swift

```diff
@@ -85,50 +84,81 @@ public class Server<InitPayload: Equatable & Codable> {
                         self.error(.invalidRequestFormat(messageType: .complete))
                         return
                     }
-                self.onOperationComplete(completeRequest.id)
+                    self.onOperationComplete(completeRequest.id)
                 case .unknown:
                     self.error(.invalidType())
+                // Addition for Datasync: Servers may now receive "next" responses from clients
+                case .next:
+                    guard let nextRequest = try? self.decoder.decode(NextRequest.self, from: data) else {
+                        self.error(.invalidRequestFormat(messageType: .next))
+                        return
+                    }
+                    do {
+                        if try nextRequest.payload.isSubscription() {
+                            self
+                                .error(.init(
+                                    "Cannot initialize a subscription within another subscription. Payload must be query or mutation",
+                                    code: .invalidRequestFormat
+                                ))
```

Comment:
Awesome! We should probably also check that we are within an active subscription (i.e. a subscription operation has begun and we are actively sending responses back to the client)

Reply:
Updated to ensure that we have initialized already. 
Per our discussion: subscription verification can now happen at the implementer level of this Server using the callbacks